lib/file_storage/local_file.rb

```diff
@@ -29,13 +32,13 @@ def save_file(path, content, _options = nil)
     end
 
     def url_for_file(path)
-      "file://#{full_path(path)}"
+      polymorphic_url('api_v0_raw', id: path)
```

Comment:
See notes above about a) baking URL structure into the DB and b) the way this causes file store implementation work (loading a file from the right location on disk) to be re-implemented outside a file store, and also means parts of the application are tied both to a particular type of storage and a particular location on disk (`LocalFile@directory` is configurable per instance, but not anymore, because it’s hardcoded in your controller).